schema/data/portal/form_json/electricity_and_heat.json

```diff
@@ -1,113 +1,145 @@
+        "type": "object",
+        "properties": {
+          "purchased": {
+            "type": "number",
+            "title": "Purchased"
+          },
+          "generatedOnSite": {
+            "type": "number",
+            "title": "Generated On-Site"
+          },
+          "consumedOnSite": {
+            "type": "number",
+            "title": "Consumed On-Site"
+          },
+          "sold": {
+            "type": "number",
+            "title": "Sold"
+          }
+        },
+        "dependencies": {
+          "generatedOnSite": {
+            "oneOf": [
+              {
+                "properties": {
+                  "generatedOnSite": { "exclusiveMinimum": 0},
+                  "onSiteEmissions": {
+                    "type": "number",
+                    "title": "Emissions from On-Site Generation (tCO2e)"
+                  }
                 }
-              ],
-              "allowAddPanel": false,
-              "allowRemovePanel": false,
-              "panelCount": 1
-            },
-            {
-              "type": "paneldynamic",
-              "name": "heat",
-              "title": "Heat (GJ)",
-              "templateElements": [
-                {
-                  "type": "text",
-                  "name": "description",
-                  "title": "Purchased",
-                  "inputType": "number"
-                },
-                {
-                  "type": "text",
-                  "name": "generatedOnsite",
-                  "startWithNewLine": false,
-                  "title": "Generated Onsite",
-                  "inputType": "number"
-                },
-                {
-                  "type": "text",
-                  "name": "consumedOnsite",
-                  "startWithNewLine": false,
-                  "title": "Consumed Onsite",
-                  "inputType": "number"
-                },
-                {
-                  "type": "text",
-                  "name": "sold",
-                  "width": "33.5%",
-                  "title": "Sold",
-                  "inputType": "number"
-                },
-                {
-                  "type": "text",
-                  "name": "quantity",
-                  "width": "66.5%",
-                  "startWithNewLine": false,
-                  "title": "Emissions from onsite generation (tCO2e)",
-                  "inputType": "number"
+              },
+              {
+                "properties": {
+                    "generatedOnSite": { "const": 0}
                 }
-              ],
-              "allowAddPanel": false,
-              "allowRemovePanel": false,
-              "panelCount": 1
-            }
-          ],
-          "allowAddPanel": false,
-          "allowRemovePanel": false,
-          "panelCount": 1
+              }
+            ]
+          }
         }
-      ]
+      },
+      "heat": {
+        "title": "Heat",
+        "type": "object",
+        "properties": {
```

Comment:
Can you use a definition (see address in the admin page) instead of repeating the fields?